test/cluster/test_aggregation.py

```diff
@@ -57,25 +60,37 @@ async def test_cancel_mapreduce(manager: ManagerClient):
             s2_mark = await s2_log.mark()
 
             # Prevent finishing local mapreduce tasks on node 2.
-            async with inject_error(manager.api, s2.ip_addr, "mapreduce_pause_dispatch_to_shards"):
-                async def do_select():
-                    # Make node 1 the supercoordinator of the mapreduce task corresponding to aggregation.
-                    # We use this timeout because it's longer than the cumulative timeout of the following
-                    # steps. For the test to be reliable, the query cannot end on its own.
-                    try:
-                        await cql.run_async(f"SELECT count(*) FROM {t} BYPASS CACHE USING TIMEOUT 600s", host=host1)
-                        pytest.fail(f"Query finished, but it wasn't supposed to")
-                    except NoHostAvailable:
-                        pass
-
-                async def wait_and_shutdown():
-                    # Make sure node 1 is the supercoordinator and sends a mapreduce task to node 2.
-                    await s1_log.wait_for(f"dispatching mapreduce_request=.* to address={host_id2}", from_mark=s1_mark, timeout=60)
-                    # Make sure that node 2 is preventing its local mapreduce task from finishing.
-                    await s2_log.wait_for("mapreduce_pause_dispatch_to_shards: waiting for message", from_mark=s2_mark, timeout=60)
-                    # Verify that the supercoordinator stops without an issue despite the ongoing mapreduce task.
+            await manager.api.enable_injection(s2.ip_addr, "mapreduce_pause_dispatch_to_shards", one_shot=False)
```

Comment:
Why did you change `with inject_error` to `manager.api.enable_injection`? This change seems unnecessary and makes the diff larger than it should be / masks some changes that you did due to indentation change.

Reply:
The injection is enabled on node 2 (receiver). 
`with inject_error` disables the injection at the end, so when `stop_node == receiver`, it causes an error because the host is no longer available.

Reply:
Ok, got it, thanks.